docs/src/how-to/programmers-guide.rst

```diff
@@ -0,0 +1,202 @@
+
+You can run a test via the ``tox ci`` environment by passing the desired test marker using ``-m <MARKER>``:
+
+.. code-block::
+
+   tox run -e ci -- -m {pre_checkin|extended|integration|unit}
+
+Note that ``--`` is used to pass options to the underlying pytest command.
+
+.. note::
+
+   By default, the ``tox run`` command runs pre-checkin tests, when no markers are specified via ``-m``.
+
+-------------------------------
+Unit tests and coverage reports
+-------------------------------
+
+All unit tests are available in ``Tensile/Tests/unit/``. A convenience command is included to add coverage reporting:
+
+.. code-block::
+
+   tox run -e unittest
+
+   # OR for 32 processes
+
+   tox run -e unittest -- -n 32
+
+By default, coverage results are dumped to the terminal. To generate reports in other formats such as HTML, use:
+
+.. code-block::
+
+   tox run -e unittest -- --cov-report=html
+
+Files and directories excluded from coverage reporting are itemized in `.coveragerc <https://github.com/ROCm/Tensile/blob/develop/.coveragerc>`_.
+
+Although, we encourage to run unit tests using ``tox`` for consistency, you can also run the tests directly using ``pytest`` for quicker feedback. For example, To run a single test named ``test_foo``, use:
+
+.. code-block::
+
+   pytest unit/test_TensileCreateLibrary.py -k "test_foo" --capture=no -v
+
+------------------
+Host library tests
+------------------
+
+Host library tests ensure that the generated libraries remain operational when called from the client code such as other libraries or applications.
+These tests are built on `gtest <https://github.com/google/googletest>`_. To run them, download the submodule first. Then, from Tensile project's root, run:
+
+.. code-block::
+
+   git submodule update --init
+
+Next, you can configure, build, and run the host library tests using any of the following:
+
+- ``tox``:
+
+  .. code-block::
+
+   tox run -e hostlibtest
+
+  .. note::
+
+   Note that the ``tox`` command wraps `invoke <https://www.pyinvoke.org/index.html>`_, a tool to manage CLI-invokable tasks. Since tox is fundamentally a Python environment manager and test runner, any reusable shell commands that fall outside its purview are managed by invoke (which are again encapsulated by tox sometimes). See `tasks.py <https://github.com/ROCm/Tensile/blob/develop/tasks.py>`_ for details.
+
+- ``invoke``:
+
+  .. code-block::
+
+   invoke hostlibtest --configure --build --run
+
+  Running the preceding command generates an executable ``TensileTests``, which can be further used to run the tests.
+
+- Manually: To build and run the tests manually, see the commands in `tasks.py <https://github.com/ROCm/Tensile/blob/develop/tasks.py>`_.
+  For advanced usage like filtering or repeating test cases, see the `gtest documentation <https://github.com/google/googletest/blob/main/docs/advanced.md>`_.
+
+===============
+Static analysis
+===============
+
```

Comment:
@bstefanuk We need some information about static analysis here. Also some sentences to explain the Python and C++ classification.

Reply:
This is WIP. For you, you can either remove it entirely or add a callout that this section is under development.

Reply:
ok.